tests/index_lifecycle.yaml

```diff
@@ -0,0 +1,85 @@
+$schema: ../../json_schemas/story.schema.yaml
+
+description: This story tests all endpoints relevant the lifecycle of an index, from creation to deletion.
+prologues: []
+epilogues:
+  - path: /books,movies,games
+    method: DELETE
+    ignore_errors: false
+chapters:
+  - synopsis: Create an index named `books` with mappings and settings.
+    path: /{index}
+    method: PUT
+    parameters:
+      index: books
+    requestBody:
+      content_type: application/json
+      payload:
+        mappings:
+          properties:
+            name:
+              type: keyword
+            age:
+              type: integer
+        settings:
+          number_of_shards: 5
+          number_of_replicas: 2
+    response:
+      status: 200
+
+  - synopsis: Create an index named `games` with default settings,
+    path: /{index}
+    method: PUT
+    parameters:
+      index: games
+    response:
+      status: 200
+
+  - synopsis: Check if the index `books` exists. It should.
+    path: /{index}
+    method: HEAD
+    parameters:
+      index: books
+    response:
+      status: 200
+
+  - synopsis: Check if the index `movies` exists. It should not.
+    path: /{index}
+    method: HEAD
+    parameters:
+      index: movies
+    response:
+      status: 404
+
```

Comment:
We also should test a `GET` that returns a 404 with details (incomplete schema below).

```
@@ -2314,6 +2322,18 @@ components:
             type: object
             additionalProperties:
               $ref: '../schemas/indices._common.yaml#/components/schemas/IndexState'
+    indices.get@404:
+      description: ''
+      content:
+        application/json:
+          schema:
+            type: object
+            properties:
+              status:
+                description: HTTP status code returned for the operation.
+                type: number
+              error:
+                $ref: '../schemas/_common.yaml#/components/schemas/ErrorCause'
     indices.get_alias@200:
       description: ''
       content:
diff --git a/tests/index_lifecycle.yaml b/tests/index_lifecycle.yaml
index 7ea4296..8e078ff 100644
--- a/tests/index_lifecycle.yaml
+++ b/tests/index_lifecycle.yaml
@@ -51,6 +51,14 @@ chapters:
     response:
       status: 404
 
+  - synopsis: Check if the index `movies` exists. It should not and return details.
+    path: /{index}
+    method: GET
+    parameters:
+      index: movies
+    response:
+      status: 404
+
```

We should have an error/warning for additional fields missing in spec, in the above example the entire error has things like `index_uuid`:

```json
{
  "error": {
    "root_cause": [
      {
        "type": "index_not_found_exception",
        "reason": "no such index [movies]",
        "index": "movies",
        "resource.id": "movies",
        "resource.type": "index_or_alias",
        "index_uuid": "_na_"
      }
    ],
    "type": "index_not_found_exception",
    "reason": "no such index [movies]",
    "index": "movies",
    "resource.id": "movies",
    "resource.type": "index_or_alias",
    "index_uuid": "_na_"
  },
  "status": 404
}
```

Reply:
Will handle the missing field definition in a followup PR